File: app/src/main/java/com/example/todo/data/TodoDatabase.java
package com.example.todo.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(version = 2, entities = {TodoItem.class}, exportSchema = false)
public abstract class TodoDatabase extends RoomDatabase {
    public abstract TodoDao todoDao();
    private volatile static TodoDatabase instance;

    public static TodoDatabase getInstance(final Context context){
        if(instance == null){
            synchronized (TodoDatabase.class){
                if (instance == null){
                    instance = Room.databaseBuilder(context.getApplicationContext(),
                            TodoDatabase.class,
                            "todo_database").build();
                }
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/example/todo/data/TodoDatabase.kt
package com.example.todo.data

import android.content.Context
import android.util.Log
import androidx.room.*
import androidx.room.migration.AutoMigrationSpec
import androidx.room.migration.Migration
import androidx.sqlite.db.SupportSQLiteDatabase

@Database(
    version = 4,
    entities = [TodoItem::class]
)
abstract class TodoDatabase : RoomDatabase() {
    abstract fun todoDao(): TodoDao

    companion object {
        private var instance: TodoDatabase? = null

        @Synchronized
        fun getInstance(context: Context): TodoDatabase {
            instance?.let {
                return it
            }
            instance = Room.databaseBuilder(
                context.applicationContext,
                TodoDatabase::class.java,
                "todo_database"
            ).build()
            return instance!!
        }

    }
}